server/services/diskUsageService.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * This service is not per rtorrent session, which is why it does not inherit
+ * `BaseService` nor have any use of the per user API ie. `getSerivce()`
+ */
+const EventEmitter = require('events');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const diskUsageServiceEvents = require('../constants/diskUsageServiceEvents');
+
+const diskUsage = {
+  linux: () =>
+    exec('df --block-size=1 --type=ext4 --portability | tail -n+2').then(({stdout}) =>
+      stdout
+        .trim()
+        .split('\n')
+        .map(disk => {
+          const [, /* fs */ size, used, avail /* pcent */, , target] = disk.split(/\s+/);
+          return {
+            size: Number.parseInt(size, 10),
+            used: Number.parseInt(used, 10),
+            avail: Number.parseInt(avail, 10),
+            target,
+          };
+        }),
+    ),
+  // win32
+  // darwin
+};
+
+const INTERVAL_UPDATE = 10000;
```

Comment:
Polling every 10 seconds seems a little aggressive to me, but I don't know how expensive `df` is. What do you think about once every 5 minutes?

Reply:
i could, but this should only happen while there is an open activity stream. and there should always only be one DiskUsageStorage instance to make sure that df is not run more with more open sockets. I don't think it should have much impact performance. I like to see my disk space clearing up after removing a torrent so it's nice from that pov

Reply:
It may however cause some excess disk usage change events and network traffic and ui updates, maybe i should set the blocksize to 1MiB...

Reply:
I'm still not crazy about the 10 second poll interval, but it seems fine for now. I assume Flood users' sessions are usually pretty short, so running every 5 seconds while their session is active hopefully won't be problematic.

If you're up for it, I think it would be better to use a long poll interval (like 5 minutes), and then manually invoke `diskUsageService.updateDisks` when disk usage is expected to have changed (e.g. after the server handles a request to remove torrents).

If you take this approach, I also suggest modifying this code such that the `updateDisks` invocation is scheduled via `setTimeout`. `updateDisks` would be responsible for clearing and setting its own delayed invocation, so that it's invoked immediately when it needs to be, but never more than once per 5 minutes.

Reply:
Along with that idea why not just make it a configurable setting in config.js or in the settings page. That way those who want quick updates have that option avaible to them.